mac

```diff
@@ -254,15 +262,29 @@ fancy_echo "nodenv configured."
 
 fancy_echo "Configuring OH-MY-ZSH..."
 
+
 # oh-my-zsh
 fancy_echo "Installing omyzsh..."
-sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
+
+if [ ! -d "$ZSH" ]; then
+  sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
+else 
+  ZSH_CUSTOM="$ZSH/custom"
+fi
+
 
 # spaceship
+SPACESHIP_ROOT="$ZSH_CUSTOM/themes/spaceship-prompt"
+
 fancy_echo "Installing spaceship..."
-git clone https://github.com/denysdovhan/spaceship-prompt.git "$ZSH_CUSTOM/themes/spaceship-prompt"
-ln -s "$ZSH_CUSTOM/themes/spaceship-prompt/spaceship.zsh-theme" "$ZSH_CUSTOM/themes/spaceship.zsh-theme"
-sed -i "" "/ZSH_THEME=*/ZSC_THEME=\"spaceship\"/g" .zshrc
+
+if [ ! -d "$SPACESHIP_ROOT" ]; then
+  git clone https://github.com/denysdovhan/spaceship-prompt.git "$SPACESHIP_ROOT"
+  ln -s "$SPACESHIP_ROOT/spaceship.zsh-theme" "$ZSH_CUSTOM/themes/spaceship.zsh-theme"
+  sed -i "" "s/^ZSH_THEME=.*/ZSH_THEME=\"spaceship\"/g" .zshrc
+else
+  git -C "$SPACESHIP_ROOT" pull
+fi
```

Comment:
It is not clear to me the intended behaviour. How is `$ZSH` set? How the user would decide to install a custom or a standard zsh?

_suggestion: you can add a block comment to explain the decision you made_

Reply:
updated in the new commit. It now uses zsh to load ohmyzsh variables